web/src/games/soupofletters/game.ts

```diff
@@ -0,0 +1,138 @@
+
+import { MAX_WORDS_IN_GAME, MAX_WORD_LEN, DRAW_AFTER_N_TIMERS, validOrientations, globalWordList } from './constants';
+import { newPuzzle, solvePuzzel, printPuzzle } from './puzzel'; 
+
+
+export interface ISingleLetter {
+    x: number;
+    y: number;
+    letter: string;
+}
+
+export interface ISolvedWord{
+    x:number;
+    y:number;
+    orientation: string;
+    word: string;
+    overlap: number;
+    solvedBy?: string;
+    letters: Array<ISingleLetter>;
+}
+
+export interface IG {
+    puzzel: Array<Array<string>>;
+    solution: Array<ISolvedWord>;
+    timeRef: number;
+    countTimerFired: number;
+}
+
+function isGameOver(G, ctx) {
+    // if timer fired too many times, end game 
+    if (G.countTimerFired >= (ctx.numPlayers * DRAW_AFTER_N_TIMERS )) { return true; } 
```

Comment:
Is this going to work on the server-side/multiplayer games? I am not sure if bgio supports this

Reply:
Cannot resolve this, beacuse I cannot start the online mode game

Reply:
You are right, this does not work. Do you have a solution for this. How can I get the number of players. 
There are 2 ways from how i see this, 
1. store number of players when in G during setup (dont know it its possible)
2. count failed attempts per player, and then check if all players failed a fixed number of times.